services/ui-src/src/utils/forms/dynamicItemFields.ts

```diff
@@ -172,3 +172,27 @@ const updatedItemChoiceList = (
   }
   return updatedChoiceList;
 };
+
+// dynamically generate fields for choices
+export const generateAnalysisMethodFields = (
+  form: FormJson,
+  items: AnyObject[]
+) => {
+  const choicesToInject = availableAnalysisMethods(items);
+  form!.fields[0]!.props!.choices[0].children[1].props.choices =
+    choicesToInject;
```

Comment:
this isn't pretty but we have done it this way before.
An alternative is to filter by the keys for the arrays, so that at least if the order changes we still end up at the right one. That said, it'd be more verbose and not necessarily nicer better to read.

At the very least I'd add a comment here indicating which fieldId it's targeting so that a developer can more easily reference it back in the naaar.json